lib/traject/config/sirsi_config.rb

```diff
@@ -2219,73 +2138,22 @@ def holdings(record, context)
-
-    if call_number_object
-      scheme = call_number_object.scheme.upcase
-      # if it's a shelved-by location, use a totally different way to get the callnumber
-      if holding.shelved_by_location?
-        if [holding.home_location, holding.current_location].include? 'SHELBYSER'
-          lopped_call_number = "Shelved by Series title"
-        else
-          lopped_call_number = "Shelved by title"
-        end
-
-        enumeration = holding.call_number.to_s[call_number_object.lopped.length..-1].strip unless holding.ignored_call_number?
-        shelfkey = lopped_call_number.downcase
-        reverse_shelfkey = CallNumbers::ShelfkeyBase.reverse(shelfkey)
-
-        call_number = [lopped_call_number, (enumeration if enumeration)].compact.join(' ') unless holding.e_call_number?
-        volume_sort = [lopped_call_number, (CallNumbers::ShelfkeyBase.reverse(CallNumbers::ShelfkeyBase.pad_all_digits(enumeration)).ljust(50, '~') if enumeration)].compact.join(' ').downcase
-      # if there's only one item in a library/home_location/call_number_type, then we use the non-lopped versions of stuff
-      elsif stuff_in_the_same_library.length <= 1
-        shelfkey = call_number_object.to_shelfkey
-        volume_sort = call_number_object.to_volume_sort
-        reverse_shelfkey = call_number_object.to_reverse_shelfkey
-        lopped_call_number = call_number_object.call_number
-      else
-        # there's more than one item in the library/home_location/call_number_type, so we lop
-        shelfkey = call_number_object.to_lopped_shelfkey == call_number_object.to_shelfkey ? call_number_object.to_shelfkey : "#{call_number_object.to_lopped_shelfkey} ..."
-        volume_sort = call_number_object.to_volume_sort
-        reverse_shelfkey = call_number_object.to_lopped_reverse_shelfkey
-        lopped_call_number = call_number_object.lopped == holding.call_number.to_s ? holding.call_number.to_s : "#{call_number_object.lopped} ..."
-
-        # if we lopped the shelfkey, or if there's other stuff in the same library whose shelfkey will be lopped to this holding's shelfkey, we need to add ellipses.
-        if call_number_object.lopped == holding.call_number.to_s && stuff_in_the_same_library.reject { |x| x.call_number.to_s == holding.call_number.to_s }.select { |x| call_number_for_holding(record, x, context).lopped == call_number_object.lopped }.any?
-          lopped_call_number += " ..."
-          shelfkey += " ..."
-        end
-      end
-    else
-      scheme = ''
-      shelfkey = ''
-      volume_sort = ''
-      reverse_shelfkey = ''
-      lopped_call_number = holding.call_number.to_s
-    end
-
-    current_location = holding.current_location
-    current_location = 'ON-ORDER' if holding.is_on_order? && holding.current_location && !holding.current_location.empty? && holding.home_location != 'ON-ORDER' && holding.home_location != 'INPROCESS'
```

Comment:
I'm pretty sure this will _never_ get set because `holding.is_on_order?` only evaluates to `true` if `holding.current_location` is _already_ set to `ON-ORDER` OR if `holding.home_location` is 'ON-ORDER' in which case the latter `holding.home_location != 'ON-ORDER'` condition will not pass. 😕